lib/src/com/hydrologis/dart_jts/index/strtree.dart

```diff
@@ -985,7 +985,7 @@ class STRtree extends AbstractSTRtree implements SpatialIndex {
     for (int j = 0; j < sliceCount; j++) {
       slices[j] = [];
       int boundablesAddedToSlice = 0;
-      while (i.moveNext() && boundablesAddedToSlice < sliceCapacity) {
+      while (boundablesAddedToSlice < sliceCapacity && i.moveNext()) {
```

Comment:
Hi, thanks for this. I am not sure I understand though. 

i.moveNext should return false, so why exactly are you turning it around? Thank you.

Reply:
after i.moveNext(), if boundablesAddedToSlice >= sliceCapacity
that make i.current missing.

Reply:
Ahhh, you are right. I wonder if this happened to me in other points. This is handled differently in java and dart. :-( 

Thanks!